### PR TITLE
refactor!: combine `automator` and `roger`

### PR DESCRIPTION
**Pull request by wodeni**
# Description

This PR combines our command-line tools `roger` and `automator` into one single CLI. 

# Implementation strategy and design decisions

* Deprecate artifact static site generation in `automator`
* Convert `roger`'s websocket code to work with ESM
* Change from `ts-node` to `tsx`
* Bumped some package versions to `3.0.0-alpha.x` for testing
* Add an example of `trio.json` in `triangle-mesh-2d`. This example uses multiple Style files for one Substance by concatenating them.

# Examples with steps to reproduce them

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have reviewed any generated registry diagram changes

# Open questions

* When running `roger` (after the `yarn link` installation) inside `packages/examples/src`, `process.cwd()` only resolves to `packages/examples`. Reading files would fail due to the incorrect path. The reason is still unclear, possibly due to symlinks (?).

### CODE REVIEW

**Review by samestep — COMMENTED** [review]
I'm guessing this wasn't meant to be marked as ready for review but here are a few random comments in case they're helpful at all (guessing not)

File: packages/roger/types.ts [hunk 1]
```diff
@@ -27,9 +27,6 @@ export interface InstanceData {
   timeTaken: TimeTaken;
   selectorMatches: [];
   optProblem: OptProblem;
-  reference: Reference;
-  ciee: number | null; // JSON turns Infinity into null
-  extra?: unknown;
```
[the remark this answers is not in the document]
Comment (wodeni) [reply packages/roger/types.ts:32]: This was specific to the artifacts static page, so I basically deprecated all metadata related to those.

**Comment by samestep** [comment]
The CI `readme` and `registry` jobs are failing; could you fix those?

File: packages/core/CHANGELOG.md
Comment (samestep) [inline packages/core/CHANGELOG.md:1]: here

File: packages/roger/CHANGELOG.md
Comment (samestep) [inline packages/roger/CHANGELOG.md:1]: and here

**Review by samestep — CHANGES_REQUESTED** [review]
Also it looks like you missed a couple other accidental `CHANGELOG.md` changes:

**Comment by samestep** [comment]
Looking at the registry diff, I see that most of them just changed the `id` prefix from `automator-` to `roger-`, but others actually have visual changes to the diagram, such as `call-directed-graph-directed-multigraph.svg`. Any idea what's causing that?

**Comment by wodeni** [comment]
> Looking at the registry diff, I see that most of them just changed the `id` prefix from `automator-` to `roger-`, but others actually have visual changes to the diagram, such as `call-directed-graph-directed-multigraph.svg`. Any idea what's causing that?

I'm really not sure. This is probably not related to the variation itself. Can't imagine `tsx` making a difference here.

**Review by samestep — CHANGES_REQUESTED** [review]
Looks great overall! Could you either update `CONTRIBUTING.md` or add back an `install-global` script to `@penrose/roger`?

File: packages/core/CHANGELOG.md [hunk 2]
```diff
@@ -176,7 +176,7 @@ See [Conventional Commits](https://conventionalcommits.org) for commit guideline
 - resolve paths for included SVGs ([#825](https://github.com/penrose/penrose/issues/825)) ([cedbf1b](https://github.com/penrose/penrose/commit/cedbf1b0f219f013a0c825e08007a2edc3b2c3bc))
 - Signed distance functions for Penrose shapes ([#979](https://github.com/penrose/penrose/issues/979)) ([1a00e4c](https://github.com/penrose/penrose/commit/1a00e4c113c8e1e308612e41528af50665d7b194))
 - Split Optimization Status Tab Into Constraints and Objectives ([#611](https://github.com/penrose/penrose/issues/611)) ([ab9eee9](https://github.com/penrose/penrose/commit/ab9eee988df39ff2afee1c8dfc11ab45b75bc7e3))
-- staged diagram generation in automator ([#610](https://github.com/penrose/penrose/issues/610)) ([3de4a31](https://github.com/penrose/penrose/commit/3de4a31543ddac80ed24274fde66d9e84304daa1))
+- staged diagram generation in roger ([#610](https://github.com/penrose/penrose/issues/610)) ([3de4a31](https://github.com/penrose/penrose/commit/3de4a31543ddac80ed24274fde66d9e84304daa1))
```
Comment (samestep) [inline packages/core/CHANGELOG.md:179]:
```suggestion
- staged diagram generation in automator ([#610](https://github.com/penrose/penrose/issues/610)) ([3de4a31](https://github.com/penrose/penrose/commit/3de4a31543ddac80ed24274fde66d9e84304daa1))
```

File: packages/roger/CHANGELOG.md
Comment (samestep) [inline packages/roger/CHANGELOG.md:1]: Can you revert your changes to this file? Unless your intention was to combine this with the `@penrose/automator` changelog and you're doing something clever here...

File: packages/roger/index.ts [hunk 3]
```diff
@@ -0,0 +1,514 @@
+import "global-jsdom/register"; // must be first
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+
+import {
```
Comment (samestep) [inline packages/roger/index.ts:5]:
```suggestion
import "global-jsdom/register"; // must be first

import yargs from "yargs";
import { hideBin } from "yargs/helpers";
import {
```

File: packages/roger/CHANGELOG.md
Comment (wodeni) [reply packages/roger/CHANGELOG.md:1]: The intention is to retain the `automator` CHANGELOG, as it make up the majority of `roger` code.

File: .prettierignore [hunk 4]
```diff
@@ -1,6 +1,5 @@
 *.hbs
-/packages/automator/out/
-/packages/automator/progs/
+/packages/roger/out/
```
Comment (samestep) [inline .prettierignore:2]: Alphabetize pls 😛

**Review by samestep — APPROVED** [review]

**Pull request merged**